pkg/cloudprovider/provider/azure/provider.go

```diff
@@ -136,9 +137,12 @@ func New(configVarResolver *providerconfig.ConfigVarResolver) cloud.Provider {
 	return &provider{configVarResolver: configVarResolver}
 }
 
-func (p *provider) getConfig(s runtime.RawExtension) (*config, *providerconfig.Config, error) {
+func (p *provider) getConfig(s v1alpha1.ProviderConfig) (*config, *providerconfig.Config, error) {
+	if s.Value == nil {
```

Comment:
Maybe `len(s.Value) == 0` would be better, or is a initialized 0-len slice equal to `nil` ?

Reply:
Nevermind, `s.Value` is not a slice